### PR TITLE
DCOS-57645: feat(endpointstab): add weburl to endpoints tab

### PR DESCRIPTION
**Pull request by Poltergeist**
This adds the weburl from a service to the endpoints tab as a click to copy function

CLOSES DCOS-57645

## Testing

Start a  package with a frontend (marathon) for example.
visit the endpoints tab on the detail page and check if the url is present there.

## Trade-offs

N/A

## Dependencies

N/A

## Screenshots

image

### CODE REVIEW

File: plugins/services/src/js/containers/service-connection/ServiceConnectionEndpointList.js [hunk 1]
```diff
@@ -214,4 +244,12 @@ ServiceConnectionEndpointList.contextTypes = {
   router: routerShape
 };
 
+function hasWebUI(service) {
+  return (
+    service instanceof Service &&
+    service.getWebURL() != null &&
```
Comment (GeorgiSTodorov) [inline plugins/services/src/js/containers/service-connection/ServiceConnectionEndpointList.js]: Why do we do "!=" here?

File: plugins/services/src/js/containers/service-connection/ServiceConnectionEndpointList.js [hunk 2]
```diff
@@ -165,6 +166,34 @@ class ServiceConnectionEndpointList extends React.Component {
     });
   }
 
+  getEndpoints(service) {
+    const webUrl =
+      service instanceof Pod ||
```
Comment (GeorgiSTodorov) [inline plugins/services/src/js/containers/service-connection/ServiceConnectionEndpointList.js]: Do pods always have web urls?

Comment (GeorgiSTodorov) [reply plugins/services/src/js/containers/service-connection/ServiceConnectionEndpointList.js]: This page isn't even shown for pods as far as I can tell, for them we use `plugins/services/src/js/containers/service-connection/ServicePodConnectionEndpointList.js`.

File: plugins/services/src/js/containers/service-connection/ServiceConnectionEndpointList.js [hunk 3]
```diff
@@ -214,4 +244,12 @@ ServiceConnectionEndpointList.contextTypes = {
   router: routerShape
 };
 
+function hasWebUI(service) {
```
Comment (GeorgiSTodorov) [inline plugins/services/src/js/containers/service-connection/ServiceConnectionEndpointList.js]: What is the point of this check? Can't we just make `getWebURL` return null or "'?

File: plugins/services/src/js/containers/service-connection/ServiceConnectionEndpointList.js (hunk 2 above)
Comment (Poltergeist) [reply plugins/services/src/js/containers/service-connection/ServiceConnectionEndpointList.js]: Good catch.

**Comment by GeorgiSTodorov** [comment]
An integration test appears to be failing.

**Comment by GeorgiSTodorov** [comment]
For some reason, the web url wasn't showing in the integration test. I removed this check for now and also rebased.

**Comment by pierrebeitz** [comment]
i rebased the fixups directly in master...

**Comment by GeorgiSTodorov** [comment]
Why do we want no fixups on master?

**Comment by pierrebeitz** [comment]
fixup-commits are not meant to be separate commits but rather should be "squashed" into the commits they are considered an amendment for because they modify stuff to be correct in the first place. 

all by themselves they're just unnecessary noise without a proper commit-messsage.

**Comment by GeorgiSTodorov** [comment]
This seems like a pretty big downside to me. I am not sure if it's worth it using them.

**Comment by pierrebeitz** [comment]
i used them to "suggest changes" here as discussed in the sync.

you would run `git rebase -i master` locally to squash them into the places they're meant to be if you like the changes and them push and merge. maybe it's worth discussing this in the sync again.

**Comment by GeorgiSTodorov** [comment]
Yes, I would be in favour of discussing this. If it's a small change, I think we can just say what we changed (also, it would be much more clear in the PR since there would be comments) and if it's a big change then it shouldn't be a fixup. If we have to rebase and push again, we would have to dismiss reviews also.

**Pull request merged**